prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -222,31 +237,40 @@ class WorkflowEngine(
         if (!isTest) queue.sendMessage(nextEvent) // Avoid race condition by doing after txn completes.
     }
 
-    fun fetchItemsThatFailed(reportFile: ReportFile): RetryItems {
+    /**
+     * Given a batched report, look at its children 'sent' reports, and gather the disposition
+     * of every item in the batched report.   If there are no children,
+     * this will return a map with all item statuses == NEVER_ATTEMPTED
+     */
+    fun fetchItemDispositions(reportFile: ReportFile): Map<Int, RedoxTransport.ResultStatus> {
         if (reportFile.nextAction != TaskAction.send) {
             throw Exception("Cannot send ${reportFile.reportId}. Its next action is ${reportFile.nextAction}")
         }
         if (reportFile.bodyFormat != Report.Format.REDOX.name) {
             throw Exception("SendFailed option only applies to REDOX reports.  This report is ${reportFile.bodyFormat}")
         }
-        val successfulItemsMap = mutableMapOf<Int, Boolean>() // <itemNumber, true = successfully sent>
+        // Track what happened to each item.
+        val itemsDispositionMap = mutableMapOf<Int, RedoxTransport.ResultStatus>().apply {
+            // Initialize all to assume nothing was every done
+            for (i in 0 until reportFile.itemCount) this[i] = RedoxTransport.ResultStatus.NEVER_ATTEMPTED
```

Comment:
This is the key line for the bug fix:  if nothing ever happened, pre-load that into the Map as a failure.   It'll stay marked as a failure unless a success was found in the item lineage history data.